src/google_apps_clj/credentials.clj

```diff
@@ -87,10 +89,41 @@
       (.setRefreshToken (:refresh-token auth-map))
       (.setTokenType (:token-type auth-map)))))
 
-(t/ann ^:no-check build-credential [GoogleCtx -> HttpRequestInitializer])
+(t/ann credential-with-scopes [GoogleCredential (t/Seq t/Str) -> GoogleCredential])
```

Comment:
I don't think we can, unfortunately.  The `GoogleCredential` class models its scopes as a `Collection<String>`.

OAuth scopes are just strings, and the set of scopes recognized by the Google API endpoints ([link](https://developers.google.com/identity/protocols/googlescopes)) is quite large and may change and is not a function of the client library, so I don't think we can or should attempt to reject invalid scopes.